### ISSUE TITLE
💡 Replace `fast-glob` by `tinyglobby`

### ISSUE THREAD

**Issue by Jayllyz** [webpro-nl/knip#768]
### Suggest an idea for this project

Most of `knip` deps come from `fast-glob`.

https://npmgraph.js.org/?q=knip

https://bundlephobia.com/package/fast-glob vs https://bundlephobia.com/package/tinyglobby

repo: https://github.com/SuperchupuDev/tinyglobby

see: https://github.com/es-tooling/module-replacements/blob/main/docs/modules/glob.md

**Comment by webpro** [comment]
Interesting. At first glance, only one feature seem to be missing: a way to stop searching when one match is found:

https://github.com/webpro-nl/knip/blob/a230582308b581d8e746d0d9672cae0a1680b68c/packages/knip/src/util/glob.ts#L62-L67

**Comment by webpro** [comment]
Related: https://github.com/webpro-nl/knip/issues/678 (would be great if an alternative contains the fix)

**Comment by SuperchupuDev** [comment]
just tested and it looks like tinyglobby doesn't have the bug described in #678

![image](https://github.com/user-attachments/assets/18af8656-137a-4d9e-a29c-497a293f7485)

**Comment by webpro** [comment]
Thanks @SuperchupuDev! Now I'd be even happier to start using tinyglobby. Even though the `globStream` feature is currently used only once in this codebase, I do like to have it available (planning to use it more). Does not have to be part of tinyglobby. For sure I'll keep an eye on tinyglobby, thanks for building it!

**Comment by SuperchupuDev** [comment]
Thanks! The problem with `globStream` is that I have to wait until `fdir` adds support for streams, there's an open issue from years ago

**Comment by webpro** [comment]
Closing this issue as part of a general cleanup to keep this project sustainable and optimize my time working on it. If you think this is inappropriate or if there is no workaround and you feel stuck, feel free to [open a new issue](./new/choose). Thanks for your understanding.
